server/config/store_config.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/tikv/pd/pkg/typeutil"
+	"io/ioutil"
+	"net/http"
+	"sync"
+)
+
+var (
+	// default region max size is 144MB
+	defaultRegionMaxSize = uint64(144)
+	// default region split size is 96MB
+	defaultRegionSplitSize = uint64(96)
+	// default region max key is 144000
+	defaultRegionMaxKey = uint64(1440000)
+	// default region split key is 960000
+	defaultRegionSplitKey = uint64(960000)
+)
+
+// StoreConfigManager is used to manage the store config.
+type StoreConfigManager struct {
+	mu     sync.RWMutex
+	config *StoreConfig
+}
+
+// NewStoreConfigManager creates a new StoreConfigManager.
+func NewStoreConfigManager() *StoreConfigManager {
+	return &StoreConfigManager{}
+}
+
+// GetStoreConfig returns the current store configuration.
+func (m *StoreConfigManager) GetStoreConfig() *StoreConfig {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+	return m.config
+}
+
+// StoreConfig is the config of store like TiKV.
+// generated by https://mholt.github.io/json-to-go/.
+// nolint
+type StoreConfig struct {
+	Coprocessor struct {
+		RegionMaxSize   string `json:"region-max-size"`
+		RegionSplitSize string `json:"region-split-size"`
+		RegionMaxKeys   int    `json:"region-max-keys"`
+		RegionSplitKeys int    `json:"region-split-keys"`
+	} `json:"coprocessor"`
+}
+
+type getBytes = func() []byte
+
+func (c *StoreConfig) getMbSize(fn getBytes, value uint64) uint64 {
+	if c == nil {
+		return value
+	}
+	body := fn()
+	var bs typeutil.ByteSize
+	if err := json.Unmarshal(body, &bs); err != nil {
+		return value
+	}
+	return bs.ToMiB()
+}
```

Comment:
It is too complex, it seems you just need a function to parse MBsize from a string ?